File: Java - DSA/src/com/algorithm/imodupsy/datastructure/hasttable/RecurringNum.java
package com.algorithm.imodupsy.datastructure.hasttable;

import java.util.HashMap;

/**
 * @author iModupsy
 * @created 06/06/2022
 */
public class RecurringNum {

    public static void main(String[] args) {
        int[] nums = {2, 5, 1, 7, 3, 5, 1, 2, 4};

        int recurringChar = firstRecurringNum(nums);
        System.out.println(recurringChar);
    }

    public static int firstRecurringNum(int[] nums) {
        HashMap<Integer, Integer> allNums = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            Integer value = allNums.get(nums[i]);
            if (value == null) {
                allNums.put(nums[i], 1);
            } else {
                allNums.put(nums[i], value + 1);
            }
        }
        System.out.println(allNums);
        return 0;

    }
}

File: javascript/dataStructure/HashTable/recurringNum.js
const nums = [2, 5, 1, 2, 3, 5, 1, 2, 4];



function firstRecurringChar(nums) {
    for (let i = 0; i < nums.length; i++) {
        for (let j = i++; j < nums.length; j++) {
            if (nums[i] === nums[j]) return nums[i]
        }
    }

    return undefined;
}

const bruteForceSolution = firstRecurringChar(nums);
console.log(bruteForceSolution);


function firstRecurringNum(nums) {
    let allNums = {};

    for (let i = 0; i < nums.length; i++) {
        if (allNums[nums[i]]) {
            return nums[i];
        } else {
            allNums[nums[i]] = i
        }   
    }

    return undefined;
}

const hastTable = firstRecurringChar(nums);
console.log(hastTable);